glib/src/object.rs

```diff
@@ -1843,6 +1899,19 @@ impl<T: ObjectType> ObjectExt for T {
         }
     }
 
+    unsafe fn connect_unsafe_id<F>(
+        &self,
+        signal_id: SignalId,
+        after: bool,
+        callback: F,
+    ) -> Result<SignalHandlerId, BoolError>
+    where
+        F: Fn(&[Value]) -> Option<Value>,
+    {
+        let signal_name = signal_id.name();
+        self.connect_unsafe(signal_name, after, callback)
```

Comment:
That's rather useless :) You already have the id here, get the name, then query based on the name again to get the id again.

Can you refactor this a bit so that `connect_unsafe()` calls `connect_unsafe_id()` after querying the signal id?

Reply:
Ah right, but we still need the signal details which we can only get by calling `SignalId::parse_name` I think?

Reply:
Unless we want the _id ones to take a `details: Option<&str>` param. This way we can parse the signal name, and call `connect_unsafe_id`. What do you think? :)

Reply:
You need to take the details as quark parameter

Reply:
In short: `connect_id()` takes `SignalId` and `Option<Quark>`, `connect()` takes a single `&str` (which is then parsed into the previous two things)

Reply:
Makes sense, i will implement that :)